File: php/features.php
<?php
declare(strict_types=1);

// Geonet SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class GeonetFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new GeonetBaseFeature();
            case "test":
                return new GeonetTestFeature();
            default:
                return new GeonetBaseFeature();
        }
    }
}

File: py/features.py
# Geonet SDK feature factory

from feature.base_feature import GeonetBaseFeature
from feature.test_feature import GeonetTestFeature


def _make_feature(name):
    features = {
        "base": lambda: GeonetBaseFeature(),
        "test": lambda: GeonetTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
